core/slice.go

```diff
@@ -480,6 +484,16 @@ func (sl *Slice) GetPendingHeader() (*types.Header, error) {
 	}
 }
 
+// GetPendingHeaderUsingSealHash is used by the miner to request the current pending header
+func (sl *Slice) GetPendingHeaderUsingSealHash(hash common.Hash) (*types.Header, error) {
+	if ph, exists := sl.phCacheSeal.Get(hash); exists {
```

Comment:
It looks like there is now a duplicate `phCache` called `phCacheSeal` and a matching set of setters/getters. Why? Why not just modify the existing `phCache` to be keyed by seal hash? It makes me itch having full duplication of this structure and its interfaces.

Reply:
Apply this criticism to the entire 'Added phCacheSeal and retrieve using sealHash` commit